packages/frontend/src/app/home.tsx

```diff
@@ -127,6 +124,20 @@ function Homepage() {
 
     connectWebSocket();
 
+    const initializeModelTypesToMetrics = async () => {
+      try {
+        setModelTypesToMetrics(await fetchMetricInfo());
+        console.log('Fetched metrics successfully');
+      } catch (error) {
+        console.error('Failed to fetch metric info:', error);
+        // Call the function again after 10 seconds
+        setTimeout(initializeModelTypesToMetrics, 10000);
+      }
+    };
+
+    // Call the initialization function
+    initializeModelTypesToMetrics();
```

Comment:
Promise not awaited - bad! There is a pattern to do what you are trying to do, but in those cases remember await/async is just syntatic sugar for (new Promise()).then(..code...), so instead of await I'd just use promise chaining with then